### ISSUE TITLE
-fsanitize=leak on ldflags should link the ASan runtime instead

### ISSUE THREAD

**Issue by primiano** [google/sanitizers#878]
TL;DR I think that when passing -fsanitize=leak to clang *during linking* that should link the ASan runtime. IIUC the LSan runtime is now part of Asan. Today -fsanitize=leak gives a link error which I find contra-intuitive.

Longer story: I started a project with sanitizers coverage. We run all sanitizers independently (i.e. on different bot configs).
Until recently we had a bot that had this config:
cflags = -fsanitize=leak
lfdlags = -fsanitize=leak
Everything was working fine with clang up to 3.8.

Then recently I tried rolling clang like chrome does (same clang version, same libcxxabi version).
When I tried to do that I got this weird error in the linker step:
/usr/bin/ld: cannot find buildtools/clang/lib/clang/6.0.0/lib/linux/libclang_rt.lsan-x86_64.a: No such file or directory

Effectively libclang_rt.lsan was not there (but the others sanitizer libs were).
After some googling I found that the lsan runtime has been moved into the asan one. So it makes sense that the libclang_rt.lsan is not there.

The fix for me was easy, I had to switch the bot to be:
cflags = -fsanitize=leak
lfdlags = -fsanitize=address   !!!

(See https://android-review.googlesource.com/c/platform/external/perfetto/+/525155/4/build/sanitizers/BUILD.gn)

However, it is IMHO a bit contra-intuitive the fact that somebody should be aware of all this.
Can we have -fsanitize=leak (passed to the linker) automatically link the asan runtime, given that the lsan runtime, as a standalone library, is no more? I don't see a good reason why clang should try to link against libclang_rt.lsan if it knows that it doesn't exist anymore.

**Comment by primiano** [comment]
+ @eugenis

**Comment by kcc** [comment]
On Linux, -fsanitize=leak links in a standalone lsan library, and it works fine. 
Maybe for some reason the standalone lsan library is not present in your installation.

**Comment by eugenis** [comment]
How do you build your toolchain? LSan is both a standalone library and part
of ASan. ToT on Linux/x86_64 has
lib/clang/6.0.0/lib/linux/libclang_rt.lsan-x86_64.a in the build directory.

**Comment by primiano** [comment]
> How do you build your toolchain?
I use the one that hwennborg&thakis bake for chromium [1].

I see maybe the problem is just with that build config. I guess in chrome that problem doesn't show up because we use asan+lsan together?

Anyhow, this seems just an issue with the chrome's toolchian build not upstream. I think we can close this.

[1] https://cs.chromium.org/chromium/src/tools/clang/scripts/update.py?type=cs&q=clang_version+f:clang+&sq=package:chromium&l=402

**Comment by kcc** [comment]
Chrome toolchain
